gateway/v10.ts

```diff
@@ -672,6 +678,57 @@ export interface GatewayChannelPinsUpdateDispatchData {
 	last_pin_timestamp?: string | null;
 }
 
+/**
+ * https://discord.com/developers/docs/monetization/entitlements#new-entitlement
+ * https://discord.com/developers/docs/monetization/entitlements#updated-entitlement
+ */
+export type GatewayEntitlementModifyDispatchData = APIEntitlement;
+
+/**
+ * https://discord.com/developers/docs/monetization/entitlements#new-entitlement
+ * https://discord.com/developers/docs/monetization/entitlements#updated-entitlement
+ */
+export type GatewayEntitlementModifyDispatch = DataPayload<
+	GatewayDispatchEvents.EntitlementCreate | GatewayDispatchEvents.EntitlementUpdate,
+	GatewayEntitlementModifyDispatchData
+>;
+
+/**
+ * https://discord.com/developers/docs/monetization/entitlements#new-entitlement
+ */
+export type GatewayEntitlementCreateDispatchData = GatewayEntitlementModifyDispatchData;
+
+/**
+ * https://discord.com/developers/docs/monetization/entitlements#new-entitlement
+ */
+export type GatewayEntitlementCreateDispatch = GatewayEntitlementModifyDispatch;
+
+/**
+ * https://discord.com/developers/docs/monetization/entitlements#updated-entitlement
+ */
+export type GatewayEntitlementUpdateDispatchData = GatewayEntitlementModifyDispatchData;
+
+/**
+ * https://discord.com/developers/docs/monetization/entitlements#updated-entitlement
+ */
+export type GatewayEntitlementUpdateDispatch = GatewayEntitlementModifyDispatch;
+
+// TODO: The payload isn't documented anywhere or tested in any specific way, so it is based on a speculation
+export type GatewayEntitlementDeleteDispatchData = Partial<APIEntitlement> & {
+	/**
+	 * ID of the entitlement
+	 */
+	id: Snowflake;
+};
```

Comment:
https://discord.com/developers/docs/topics/gateway-events#entitlement-delete

```suggestion
export type GatewayEntitlementDeleteDispatchData = GatewayEntitlementModifyDispatchData;
```

`GatewayDispatchPayload`, `GatewayEntitlementModifyDispatch`, and `GatewayEntitlementModifyDispatchData` also need updating (comments too)